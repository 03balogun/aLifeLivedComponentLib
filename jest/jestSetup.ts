/* eslint-disable no-undef */
jest.useFakeTimers('legacy'); // Stops reference error from appearing
