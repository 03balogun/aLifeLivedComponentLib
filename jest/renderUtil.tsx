import React from 'react';
import { ThemeProvider } from '../src/themes/ThemeProvider';
import { render as rtlRender } from '@testing-library/react-native';

function render(ui, { ...renderOptions } = {}) {
  function Wrapper({ children }) {
    return <ThemeProvider>{children}</ThemeProvider>;
  }

  return rtlRender(ui, { wrapper: Wrapper, ...renderOptions });
}

// re-export everything for convenience
export * from '@testing-library/react-native';

// override render method
export { render };
