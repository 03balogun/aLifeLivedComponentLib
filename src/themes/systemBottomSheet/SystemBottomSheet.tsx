import React, { useState } from 'react';
import { StyleSheet } from 'react-native';
import { Animated, Dimensions } from 'react-native';
import { Box } from '../../themes/Box';
import { Button } from '../../components/button/Button';
import { AnimatedBox } from '../../themes/animatedBox/AnimatedBox';

type BottomSheetType = {
  isVisible: boolean;
  onClose: () => void;
  children: React.ReactNode;
};

export const SystemBottomSheet: React.FC<BottomSheetType> = ({
  isVisible,
  onClose,
  children,
}) => {
  const animatedValue = new Animated.Value(isVisible ? 1 : 0);
  const screenHeight = Dimensions.get('window').height;
  const [animation] = useState(animatedValue);

  const backdrop = {
    transform: [
      {
        translateY: animation.interpolate({
          inputRange: [0, 0.01],
          outputRange: [screenHeight, 0],
          extrapolate: 'clamp',
        }),
      },
    ],
    opacity: animation.interpolate({
      inputRange: [0.01, 0.5],
      outputRange: [0, 1],
      extrapolate: 'clamp',
    }),
  };

  const slideUp = {
    transform: [
      {
        translateY: animation.interpolate({
          inputRange: [0.01, 1],
          outputRange: [0, -1 * screenHeight],
          extrapolate: 'clamp',
        }),
      },
    ],
  };

  return isVisible ? (
    <Box
      position="absolute"
      right={0}
      left={0}
      bottom={0}
      top={0}
      justifyContent="flex-end"
      backgroundColor="transparent"
    >
      <AnimatedBox
        style={[StyleSheet.absoluteFill, slideUp, backdrop]}
        backgroundColor="surfaceVisible"
      >
        <Box
          justifyContent="flex-end"
          flex={1}
          alignContent="center"
          alignItems="center"
          paddingBottom="s"
          borderTopLeftRadius="large"
          borderTopRightRadius="large"
          paddingTop="l"
        >
          {children}
          <Button
            title="Close"
            onPress={() => onClose()}
            backgroundColor="surface"
            marginTop="l"
            marginBottom="m"
          />
        </Box>
      </AnimatedBox>
    </Box>
  ) : null;
};
