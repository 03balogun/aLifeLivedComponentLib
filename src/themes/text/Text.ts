import { createText } from '@shopify/restyle';
import { Theme } from '../theme';

export const textVariants = {
  h1: {
    fontSize: 32,
    lineHeight: 38,
    color: 'contrastHighest',
  },
  h2: {
    fontSize: 24,
    lineHeight: 30,
    color: 'contrastHighest',
  },
  h3: {
    fontSize: 16,
    lineHeight: 24,
    color: 'contrastHighest',
  },
  body: {
    fontSize: 16,
    lineHeight: 20,
    color: 'contrastHighest',
  },
  caption: {
    fontSize: 14,
    lineHeight: 18,
    color: 'contrastLow',
  },
  default: {
    fontSize: 16,
    lineHeight: 20,
    color: 'contrastHighest',
  },
};

export const Text = createText<Theme>();
