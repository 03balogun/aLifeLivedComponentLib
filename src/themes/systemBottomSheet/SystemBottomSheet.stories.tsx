import React from 'react';
import { storiesOf } from '@storybook/react-native';
import { action } from '@storybook/addon-actions';

// Component
import { SystemBottomSheet } from './SystemBottomSheet';

storiesOf('Bottom sheet', module)
  .addDecorator(Story => Story())
  .add('Default bottom sheet with no children', () => (
    <SystemBottomSheet isVisible={true} onClose={action('On close fired')}>
      {}
    </SystemBottomSheet>
  ));
