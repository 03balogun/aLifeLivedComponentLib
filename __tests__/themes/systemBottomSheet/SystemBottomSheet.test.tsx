import React from 'react';
import { SystemBottomSheet } from '../../../src/themes/systemBottomSheet/SystemBottomSheet';
import { render } from '../../../jest/renderUtil';
import { Text } from '../../../src/themes/text/Text';

describe('themes >> systemBottomSheet', () => {
  describe('Should render systemBottomSheet box with no children', () => {
    test('Render snapshot correctly', () => {
      const { toJSON } = render(
        <SystemBottomSheet isVisible={true}>{}</SystemBottomSheet>
      );
      expect(toJSON()).toMatchSnapshot();
    });

    test('Should find close button', () => {
      const { getByText } = render(
        <SystemBottomSheet isVisible={true}>{}</SystemBottomSheet>
      );
      expect(getByText('Close')).not.toBeNull();
    });
  });

  describe('Should NOT render systemBottomSheet', () => {
    test('Should not render bottom sheet', () => {
      const { toJSON } = render(
        <SystemBottomSheet isVisible={false}>{}</SystemBottomSheet>
      );
      expect(toJSON()).toMatchSnapshot();
    });
  });

  describe('Should render systemBottomSheet with children', () => {
    test('Should render bottom sheet with text', () => {
      const { toJSON } = render(
        <SystemBottomSheet isVisible={true}>
          <Text>Hi there</Text>
        </SystemBottomSheet>
      );
      expect(toJSON()).toMatchSnapshot();
    });

    test('Should find text', () => {
      const { getByText } = render(
        <SystemBottomSheet isVisible={true}>
          <Text>Hi there</Text>
        </SystemBottomSheet>
      );
      expect(getByText('Hi there')).not.toBeNull();
    });

    test('Should find close button', () => {
      const { getByText } = render(
        <SystemBottomSheet isVisible={true}>{}</SystemBottomSheet>
      );
      expect(getByText('Close')).not.toBeNull();
    });
  });
});
