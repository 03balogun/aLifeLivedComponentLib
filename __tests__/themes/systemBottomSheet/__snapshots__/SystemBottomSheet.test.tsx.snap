// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`themes >> systemBottomSheet Should NOT render systemBottomSheet Should not render bottom sheet 1`] = `null`;

exports[`themes >> systemBottomSheet Should render systemBottomSheet box with no children Render snapshot correctly 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "rgba(0,0,0,.5)",
        "bottom": 0,
        "display": "flex",
        "justifyContent": "flex-end",
        "left": 0,
        "position": "absolute",
        "right": 0,
        "top": 0,
      },
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "alignContent": "center",
          "alignItems": "center",
          "borderTopLeftRadius": 16,
          "borderTopRightRadius": 16,
          "display": "flex",
          "flex": 1,
          "justifyContent": "flex-end",
          "paddingBottom": 8,
          "paddingTop": 24,
        },
      ]
    }
  >
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "#f4f5f5",
          "borderRadius": 8,
          "marginBottom": 24,
          "marginTop": 24,
          "opacity": 1,
          "padding": 16,
          "width": "90%",
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#000000",
              "display": "flex",
              "fontSize": 16,
              "fontWeight": "600",
              "lineHeight": 20,
              "textAlign": "center",
            },
          ]
        }
      >
        Close
      </Text>
    </View>
  </View>
</View>
`;

exports[`themes >> systemBottomSheet Should render systemBottomSheet with children Should render bottom sheet with text 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "rgba(0,0,0,.5)",
        "bottom": 0,
        "display": "flex",
        "justifyContent": "flex-end",
        "left": 0,
        "position": "absolute",
        "right": 0,
        "top": 0,
      },
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "alignContent": "center",
          "alignItems": "center",
          "borderTopLeftRadius": 16,
          "borderTopRightRadius": 16,
          "display": "flex",
          "flex": 1,
          "justifyContent": "flex-end",
          "paddingBottom": 8,
          "paddingTop": 24,
        },
      ]
    }
  >
    <Text
      style={
        Array [
          Object {
            "display": "flex",
          },
        ]
      }
    >
      Hi there
    </Text>
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "#f4f5f5",
          "borderRadius": 8,
          "marginBottom": 24,
          "marginTop": 24,
          "opacity": 1,
          "padding": 16,
          "width": "90%",
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#000000",
              "display": "flex",
              "fontSize": 16,
              "fontWeight": "600",
              "lineHeight": 20,
              "textAlign": "center",
            },
          ]
        }
      >
        Close
      </Text>
    </View>
  </View>
</View>
`;
